src/app/components/swap-panel-component.js

```diff
@@ -6,14 +6,29 @@ var csjs = require('csjs-inject')
 // const styles = styleguide.chooser()
 
 class SwapPanelComponent {
-  constructor (appStore) {
+  constructor (name, appStore, appManager, opt) {
+    this.name = name
+    this.opt = opt
     this.store = appStore
     // list of contents
     this.contents = {}
     // name of the current displayed content
     this.currentNode
 
-    this.store.event.on('activate', (name) => { })
+    appManager.event.on('pluginNeedsLocation', (profile, domEl) => {
+      if ((profile.prefferedLocation === this.name) || (!profile.prefferedLocation && opt.default)) {
+        this.add(profile.name, domEl)
+      }
+    })
+
+    this.store.event.on('activate', (name) => {
+      let item = this.store.get(name)
```

Comment:
Maybe `const { profile, api } = this.store.get(name)` ?
Or `entry` instead of `item` if you don't want to use destructuration.

Reply:
sure